src/server/plugins/register-form/steps/home-address-lookup-choose.js

```diff
@@ -25,26 +25,61 @@ function getHandlerFactory(prevSteps) {
     const stepData = _.get(request, `state.data.${key}`, {});
     const prevStep = getPrevStep(prevSteps, request.state.data, request);
     const template = 'register-form/address-choose-step';
-    getAddresses(postcodeData.postcode)
-      .then(addressList => {
-        const addressOptions = _.map(addressList, item => {
-          return {label: formatAddressForDisplay(item), value: item};
-        });
-        return reply.view(template, {
-          fields: getFieldData(schema),
-          data: request.state.data,
-          stepData,
-          title,
-          prevStep,
-          addressOptions
-        });
-      })
-      .catch(err => {
-        throw err;
+    let addressError = false;
+    let inputValue = '';
+    callLookup(postcodeData.postcode, postcodeData.houseNumber)
+      .then(addressOptions => {
+        if(addressOptions.length ===0){
+          addressError = true;
+          callLookup(postcodeData.postcode)
+          .then(addressOptions => {
+            reply.view(template, {
+              fields: getFieldData(schema),
+              data: request.state.data,
+              stepData,
+              title,
+              prevStep,
+              addressOptions,
+              error: addressError,
+              postcode: postcodeData.postcode,
+              housenumber: postcodeData.houseNumber
+            });
+          });
+        }
+        else{
```

Comment:
Feels like there might be a few JS style errors that the linter should pick up. Maybe look at addressing things like new space before and after parenthesis and using one true brace style.